lib/streamlit/elements/write.py

```diff
@@ -0,0 +1,214 @@
+            security. For more information, see:
+
+            https://github.com/streamlit/streamlit/issues/152
+
+            **Also note that `unsafe_allow_html` is a temporary measure and may be
+            removed from Streamlit at any time.**
+
+            If you decide to turn on HTML anyway, we ask you to please tell us your
+            exact use case here:
+            https://discuss.streamlit.io/t/96 .
+
+            This will help us come up with safe APIs that allow you to do what you
+            want.
+
+        Example
+        -------
+
+        Its simplest use case is to draw Markdown-formatted text, whenever the
+        input is a string:
+
+        >>> write('Hello, *World!* :sunglasses:')
+
+        .. output::
+        https://share.streamlit.io/0.50.2-ZWk9/index.html?id=Pn5sjhgNs4a8ZbiUoSTRxE
+        height: 50px
+
+        As mentioned earlier, `st.write()` also accepts other data formats, such as
+        numbers, data frames, styled data frames, and assorted objects:
+
+        >>> st.write(1234)
+        >>> st.write(pd.DataFrame({
+        ...     'first column': [1, 2, 3, 4],
+        ...     'second column': [10, 20, 30, 40],
+        ... }))
+
+        .. output::
+        https://share.streamlit.io/0.25.0-2JkNY/index.html?id=FCp9AMJHwHRsWSiqMgUZGD
+        height: 250px
+
+        Finally, you can pass in multiple arguments to do things like:
+
+        >>> st.write('1 + 1 = ', 2)
+        >>> st.write('Below is a DataFrame:', data_frame, 'Above is a dataframe.')
+
+        .. output::
+        https://share.streamlit.io/0.25.0-2JkNY/index.html?id=DHkcU72sxYcGarkFbf4kK1
+        height: 300px
+
+        Oh, one more thing: `st.write` accepts chart objects too! For example:
+
+        >>> import pandas as pd
+        >>> import numpy as np
+        >>> import altair as alt
+        >>>
+        >>> df = pd.DataFrame(
+        ...     np.random.randn(200, 3),
+        ...     columns=['a', 'b', 'c'])
+        ...
+        >>> c = alt.Chart(df).mark_circle().encode(
+        ...     x='a', y='b', size='c', color='c', tooltip=['a', 'b', 'c'])
+        >>>
+        >>> st.write(c)
+
+        .. output::
+        https://share.streamlit.io/0.25.0-2JkNY/index.html?id=8jmmXR8iKoZGV4kXaKGYV5
+        height: 200px
+
+        """
+        # Import dynamically, to resolve circular dependency
+        from streamlit.delta_generator import DeltaGenerator
+
+        # Cast self to DeltaGenerator, to resolve mypy type issues
+        dg = cast(DeltaGenerator, self)
+        try:
+            string_buffer = []  # type: List[str]
+            unsafe_allow_html = kwargs.get("unsafe_allow_html", False)
+
+            # This bans some valid cases like st.foo.write("a", "b"), BUT:
+            # 1) such cases are rare, 2) this is simple to understand, and
+            # 3) this will be removed once we have st.container()
+            if not dg._is_top_level and len(args) > 1:
```

Comment:
Should we add a test for this scenario?

Reply:
And just making sure, `st.foo.write` Is there a legitimate use case where something other than `sidebar` exists? Or is this just preventative for future changes?

Reply:
"foo" here was my (confusing) shorthand for something like `empty = st.empty(); empty.write("a", "b")`. I'll spell that out more explicitly in this comment.

And there is a test! It's the second part of `test_empty` -- that was probably unclear based on how I worded this comment, apologies xP

Reply:
Ah gotcha. I was looking for a testing for an exception being thrown instead,. Carry on 😊